crates/rpc/rpc/src/eth/helpers/types.rs

```diff
@@ -20,11 +21,13 @@ where
 {
     type Transaction = <Ethereum as Network>::TransactionResponse;
 
+    type TransactionError = EthApiError;
```

Comment:
this can be the `Infallible` type right?

Reply:
it can, and was initially set to infallible by @lakshya-sky . however it became complicated converting the error to an `EthApiError` and `OpEthApiError` in order to propagate it, it required explicit trait bounds for the conversion all over the place and got very messy

Reply:
hmmm, I see, I'm assuming it wasn't as easy as unwrapping when this was called at a higher level?